coresdk/src/coresdk/text.cpp

```diff
@@ -164,6 +164,47 @@ namespace splashkit_lib
         return get_font_style(font_named(name));
     }
 
+    string get_system_font_path()
+    {
+        string base_fp = base_fs_path();
+
+        #if __linux__
+            base_fp += "usr/share/fonts";
+        #elif WINDOWS
+            base_fp += "Windows\Fonts";
+        #else
+            base_fp += "System/Library/Fonts";
+        #endif
+
+        return base_fp;
+    }
+
+    string find_system_font_path(string name)
+    {
+#ifdef WINDOWS
+#define PATH_SEP "\\"
+#else
+#define PATH_SEP "/"
+#endif
+
+        vector<string> files;
+        scan_dir_recursive(get_system_font_path(), files);
+
+        transform(name.begin(), name.end(), name.begin(), ::tolower);
+
+        for (size_t i=0; i<files.size(); ++i)
+        {
+            int fi = files[i].find_last_of(PATH_SEP) +1;
+            int fd = files[i].find_last_of('.') - fi;
+            string file_name = files[i].substr(fi, fd);
+            transform(file_name.begin(), file_name.end(), file_name.begin(), ::tolower);
```

Comment:
Do we need to convert them all to lowercase and strip the file extensions? 
This tripped me up when testing as I expected font case and file extension to be preserved in the same manor as how font loading currently works.

Reply:
You don't need to, but it's worth considering that SK is cross platform, and different platforms ship with different fonts formats (and thus extensions) so it was an attempt to increase cross-platform compatibility. E.g. I think Windows ships with `TTF` fonts, whereas macOS uses `TTC`.

Regarding the casing, Windows filesystems are case insensitive [0] whereas Unix based systems can be case sensitive [1].

For example, using the font name `ARIAL.TTF` might work on Windows, but it won't work on macOS / Linux. The macOS font might be named `Arial.ttc` (differing extension) and will be case sensitive, and Linux might be `Arial.ttf` (case sensitive)

[0] - https://superuser.com/questions/165975/are-all-versions-of-windows-case-insensitive
[1] - https://unix.stackexchange.com/questions/358395/does-the-unix-standard-require-case-sensitive-filesystems

Reply:
Should we also convert the font name the user provides to lowercase as well as I think that might help to reduce potential confusion?

The file extension problem is a tricky one. Technically there’s the possibility for the same font to be present using multiple file formats. Although I believe that splashkit only supports loading of some font formats and therefore this shouldn’t be much of an issue so long as the file extension is checked to ensure it can be loaded. 

There’s still the issue of consistency with the existing font loading functions. A potential solution might be to also strip the file extension from the user input?

Reply:
Yep agreed on all fronts. Normalize user supplied font name as well as system files that were scanned (lowercase everything). Check for a direct match on the file extension, and perhaps if that fails, fall back check for a matching font name with a different extension.

I think @macite mentioned it's also worth caching this process too, so that each time a font is loaded, the entire system font directory isn't recursively scanned.